### ISSUE TITLE
inline/baseline for feynman environment in equations

### ISSUE THREAD

**Issue by pseth** [JP-Ellis/tikz-feynman#32]
Hi again,

I am trying to create this equation with a feynman diagram in it. I can't use \feynmandiagram as I can't seem to get two loops from independent vertices (or from the same vertex for that matter) so I have decided to do this:

```
\begin{equation}
Z = 1 + U 
  \begin{tikzpicture}
    \begin{feynman}
      \vertex[dot] (a) {};
      \draw (a) arc [start angle=270, end angle=-90, radius=0.3cm];
      \draw (a) arc [start angle=90, end angle=-270, radius=0.3cm];
    \end{feynman}
  \end{tikzpicture}
    + \ldots
\end{equation}
```

However, I can't get the equation to line up with the vertex using either `\begin{feynman}[inline=(a)]` or `\begin{feynman}[baseline=(a)]`. I understood that the same arguments to \feynmandiagram are applicable to the feynman environment. Am I mistaken?

Thanks again!

**Comment by JP-Ellis** [comment]
Hi,

You are looking at a somewhat subtle point here, which I perhaps should make a little clearer in the documentation, though it only occurs in this particular context (as far as I know).

The `inline` style uses the `baseline` style internally so that the resulting image has the appropriate baseline; however, the `baseline` style only works on the whole Ti*k*Z picture and thus has to be given as an optional argument after `\begin{tikzpicture}`.  When you use `\tikzfeynman[inline=(a)]`, it ensures that the Ti*k*Z environment it creates has the appropriate `baseline` style and on the surface it's all quite simple.

This changes when you use `\begin{feynman}` as any style applied to that environment only affects that environment.  This means that although `\begin{feynman}[inline=(a)]` will try and use the `baseline` style, it won't have an effect because the baseline of the whole Ti*k*Z picture will remain unchanged.

So what you want is:
```
\documentclass{standalone}

\usepackage{amsmath}
\usepackage[compat=1.1.0]{tikz-feynman}

\begin{document}
\begin{equation*}
  Z = 1 + U
  \begin{tikzpicture}[baseline=(a)]
    \begin{feynman}[inline=(a)]
      \vertex[dot] (a) {};
      \draw (a) arc [start angle=270, end angle=-90, radius=0.3cm];
      \draw (a) arc [start angle=90, end angle=-270, radius=0.3cm];
    \end{feynman}
  \end{tikzpicture}
  + \ldots
\end{equation*}
\end{document}
```

![1](https://cloud.githubusercontent.com/assets/3196162/17627496/c863d642-60f4-11e6-9f9a-a019129fc503.png)

This (to me) doesn't look exactly right yet... it would be nice if the dot could line up with the plus sign which is actually above the baseline. This can be achieved with:

```
\RequirePackage{luatex85}
\documentclass{standalone}

\usepackage{amsmath}
\usepackage[compat=1.1.0]{tikz-feynman}

\begin{document}
\begin{equation*}
  Z = 1 + U
  \begin{tikzpicture}[baseline=-\the\dimexpr\fontdimen22\textfont2\relax]
    \begin{feynman}[inline=(a)]
      \vertex[dot] (a) {};
      \draw (a) arc [start angle=270, end angle=-90, radius=0.3cm];
      \draw (a) arc [start angle=90, end angle=-270, radius=0.3cm];
    \end{feynman}
  \end{tikzpicture}
  + \ldots
\end{equation*}
\end{document}
```
![2](https://cloud.githubusercontent.com/assets/3196162/17627494/c5b6eb78-60f4-11e6-9405-2565a8f2193a.png)

**Comment by pseth** [comment]
Interesting, thanks for the speedy reply and solution.

By the way, a separate point not related to your package, but still useful to someone with this question; I noticed that in the standalone mode I can't get the maths fonts to render (even after importing all the ams* packages). I found two solutions:
* change to article mode or 
* add the `preview` argument: change `\documentclass{standalone}` to `\documentclass[preview]{standalone}` (see http://tex.stackexchange.com/questions/50162/how-to-make-a-standalone-document-with-one-equation).
